src/mlpack/methods/ann/layer/sum_reduce.hpp

```diff
@@ -0,0 +1,106 @@
+/**
+ * @file methods/ann/layer/add_reduce.hpp
+ * @author Andrew Furey
+ *
+ * Definition of the SumReduce class sums inputs along a given axis.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_ADD_REDUCE_HPP
+#define MLPACK_METHODS_ANN_LAYER_ADD_REDUCE_HPP
+
+#include <mlpack/prereqs.hpp>
+#include "layer.hpp"
+
+namespace mlpack {
+
+/**
+ * Implementation of the SumReduce layer. The SumReduce module sums its
+ * incoming data along a given axis.
+ *
+ * @tparam MatType Matrix representation to accept as input and use for
+ *    computation.
+ */
+template<typename MatType>
+class SumReduceType : public Layer<MatType>
```

Comment:
With #3974 merged, can we change this to `SumReduce` and remove the convenience typedef below?